pilot/pkg/config/clusterregistry/secretcontroller.go

```diff
@@ -199,12 +220,26 @@ func addMemberCluster(s *corev1.Secret, cs *ClusterStore) {
 				Namespace: s.ObjectMeta.Namespace,
 			},
 		}
-		cs.clientConfigs[s.ObjectMeta.Name] = *clientConfig
-		cs.clusters = append(cs.clusters, &cluster)
+		c.cs.clientConfigs[s.ObjectMeta.Name] = *clientConfig
+		c.cs.clusters = append(c.cs.clusters, &cluster)
+		_, client, _ := kube.CreateInterfaceFromClusterConfig(clientConfig)
+		kubectl := kube.NewController(client, kube.ControllerOptions{
+			WatchedNamespace: c.watchedNamespace,
+			ResyncPeriod:     c.resyncInterval,
+			DomainSuffix:     c.domainSufix,
+		})
+		c.serviceController.AddRegistry(
+			aggregate.Registry{
+				Name:             serviceregistry.KubernetesRegistry,
+				ClusterID:        GetClusterID(&cluster),
+				ServiceDiscovery: kubectl,
+				ServiceAccounts:  kubectl,
+				Controller:       kubectl,
+			})
```

Comment:
now that registry may be added at runtime, proper locking has to be applied for the list of registries.